### PR TITLE
Feature/event bus

### PR DESCRIPTION
**Pull request by zhenlei520**

### CODE REVIEW

File: src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IProcessor.cs [hunk 1]
```diff
@@ -4,10 +4,5 @@ public interface IProcessor
 {
     Task ExecuteAsync(CancellationToken stoppingToken);
 
-    Task SleepAsync();
-
-    /// <summary>
-    /// The time to rest after the task is executed, in milliseconds
-    /// </summary>
-    int SleepTime { get; }
+    Task DelayAsync(int delay);
```
Comment (doddgu) [inline src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IProcessor.cs:7]: If the parameter name does not specify a unit, add a comment

File: src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IntegrationEventHostedService.cs [hunk 2]
```diff
@@ -2,19 +2,21 @@
...
     {
         _logger = logger;
         _processingServer = processingServer;
     }
 
     protected override Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        _logger.LogDebug("----- IntegrationEvent background task is starting");
-        
+        Task.Delay(2000, stoppingToken);
```
Comment (doddgu) [inline src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IntegrationEventHostedService.cs:16]: Why delay?

File: src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/README.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 ## IntegrationEventBus
 
-Example：
+Example�
```
Comment (doddgu) [inline src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/README.md:5]: Messy code

File: src/DDD/MASA.Contrib.DDD.Domain.Repository.EF/Repository.cs [hunk 4]
```diff
@@ -94,6 +94,11 @@ public override Task CommitAsync(CancellationToken cancellationToken = default)
         CancellationToken cancellationToken = default)
         => _context.Set<TEntity>().Where(predicate).FirstOrDefaultAsync(cancellationToken);
 
+    public override async Task<bool> AnyAsync(
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/DDD/MASA.Contrib.DDD.Domain.Repository.EF/Repository.cs:97]: processed

File: src/Dispatcher/MASA.Contrib.Dispatcher.Events/Internal/Dispatch/DispatcherBase.cs [hunk 5]
```diff
@@ -97,15 +97,15 @@ public async Task PublishEventAsync<TEvent>(IServiceProvider serviceProvider, TE
...
             {
                 if (failureLevels != FailureLevels.Ignore)
                 {
                     throw ex;
                 }
-                logger.LogWarning("----- Publishing event {@Event} rollback error ignored: message id: {messageId} -----", @event, @event.Id);
+                logger?.LogWarning("----- Publishing event {@Event} rollback error ignored: message id: {messageId} -----", @event, @event.Id);
                 return Task.CompletedTask;
             });
         }
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/Dispatcher/MASA.Contrib.Dispatcher.Events/Internal/Dispatch/DispatcherBase.cs:111]: ok

File: src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/README.md (hunk 3 above)
Comment (zhenlei520) [reply src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/README.md:5]: processed

File: src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IntegrationEventHostedService.cs (hunk 2 above)
Comment (zhenlei520) [reply src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IntegrationEventHostedService.cs:16]: Deleted, originally wanted to delay the execution of background tasks

File: src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IProcessor.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IProcessor.cs:7]: processed

Comment (doddgu) [reply src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IProcessor.cs:12]: Why not milliseconds or `TimeSpan`?

File: src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/Processor/ProcessorBase.cs [hunk 6]
```diff
@@ -4,11 +4,11 @@ public abstract class ProcessorBase : IProcessor
 {
     public abstract Task ExecuteAsync(CancellationToken stoppingToken);
 
-    /// <summary>
-    /// Easy to switch between background tasks
-    /// </summary>
-    /// <param name="delay"></param>
-    /// <returns></returns>
+    // /// <summary>
```
Comment (doddgu) [inline src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/Processor/ProcessorBase.cs:7]: Comment?

File: src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IProcessor.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/IProcessor.cs:12]: This is changed to millseconds, but the default unit of time configured in DispatcherOptions is seconds. I think background tasks can be used in most scenarios in seconds.

File: src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/Processor/ProcessorBase.cs (hunk 6 above)
Comment (zhenlei520) [reply src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/Processor/ProcessorBase.cs:7]: processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**